### PR TITLE
Docs/component docs updates

### PR DESCRIPTION
**Pull request by krissvaa**
Document Grid Pro component theme variants `highlight-read-only-cells` and  `highlight-editable-cells` with examples.

### CODE REVIEW

File: articles/ds/components/grid-pro/index.asciidoc [hunk 1]
```diff
@@ -181,34 +181,52 @@ include::{root}/src/main/java/com/vaadin/demo/component/gridpro/GridProPreventSa
 ----
 --
 
-== Styling Editable Cells
+== Distinguish editable / read-only cells
```
Comment (jouni) [inline articles/ds/components/grid-pro/index.asciidoc:184]:
```suggestion
== Distinguish Editable / Read-Only Cells
```

**Review by jouni — COMMENTED** [review]
Headings should be in "Title Case".

File: articles/ds/components/basic-layouts/index.asciidoc [hunk 2]
```diff
@@ -300,11 +300,6 @@ include::{root}/src/main/java/com/vaadin/demo/component/basiclayouts/BasicLayout
 ----
 --
 
-.CAUTION: Combining spacing and alignment
```
Comment (jouni) [inline articles/ds/components/basic-layouts/index.asciidoc:303]: Nice catch, that we don't need this anymore!

Comment (krissvaa) [reply articles/ds/components/basic-layouts/index.asciidoc:303]: That all @rolfsmeds work 🤝

**Review by dougjewsbury — COMMENTED** [review]
AsciiDoc looks good, apart from the issue about using title case for headings.

**Review by jouni — APPROVED** [review]

File: frontend/demo/component/gridpro/grid-pro-highlight-read-only-cells.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <vaadin-grid-pro theme="highlight-read-only-cells" .items="${this.items}">
+      <!-- end::snippet[] -->
+        <vaadin-grid-column path="firstName"> </vaadin-grid-column>
+        <vaadin-grid-column path="lastName"> </vaadin-grid-column>
+        <vaadin-grid-column path="membership"></vaadin-grid-column>
+        <vaadin-grid-pro-edit-column
+          path="email"
+          header="Email (Editable)"
+        ></vaadin-grid-pro-edit-column>
+      <!-- tag::snippet[] -->
```
Comment (DiegoCardoso) [inline frontend/demo/component/gridpro/grid-pro-highlight-read-only-cells.ts:41]:
```suggestion
        <!-- end::snippet[] -->
        <vaadin-grid-column path="firstName"> </vaadin-grid-column>
        <vaadin-grid-column path="lastName"> </vaadin-grid-column>
        <vaadin-grid-column path="membership"></vaadin-grid-column>
        <vaadin-grid-pro-edit-column
          path="email"
          header="Email (Editable)"
        ></vaadin-grid-pro-edit-column>
        <!-- tag::snippet[] -->
```

File: frontend/demo/component/gridpro/grid-pro-highlight-editable-cells.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      <vaadin-grid-pro theme="highlight-editable-cells" .items="${this.items}">
+      <!-- end::snippet[] -->
+        <vaadin-grid-column path="firstName"> </vaadin-grid-column>
+        <vaadin-grid-column path="lastName"> </vaadin-grid-column>
+        <vaadin-grid-column path="membership"></vaadin-grid-column>
+        <vaadin-grid-pro-edit-column
+          path="email"
+          header="Email (Editable)"
+        ></vaadin-grid-pro-edit-column>
+      <!-- tag::snippet[] -->
```
Comment (DiegoCardoso) [inline frontend/demo/component/gridpro/grid-pro-highlight-editable-cells.ts:41]:
```suggestion
        <!-- end::snippet[] -->
        <vaadin-grid-column path="firstName"> </vaadin-grid-column>
        <vaadin-grid-column path="lastName"> </vaadin-grid-column>
        <vaadin-grid-column path="membership"></vaadin-grid-column>
        <vaadin-grid-pro-edit-column
          path="email"
          header="Email (Editable)"
        ></vaadin-grid-pro-edit-column>
        <!-- tag::snippet[] -->
```

**Comment by jouni** [comment]
🤞

**Pull request merged**